### ISSUE TITLE
Generated images are corrupt on Mac

### ISSUE THREAD

**Issue by cgrinds** [ecm85/GtR#1]
Using v1.3.1-osx.zip
```
mkdir Gtr
mv v1.3.1-osx.zip Gtr
cd Gtr/
unzip v1.3.1-osx.zip
ll
-rw-r--r--@ 1   40M Oct 17 01:51 GtR
-rw-r--r--@ 1  263B Oct 17 01:46 GtRConfig.json
drwxr-xr-x@ 7  224B Oct 17 15:11 Images/
-rw-r--r--@ 1   59K Oct 17 00:52 NeuzeitGro-BolModified.ttf
-rw-r--r--@ 1   59K Sep 29 15:34 NeuzeitGro-RegModified.ttf
-rw-r--r--@ 1  5.0K Oct 17 00:46 OrderCardData.txt
-rw-r--r--@ 1  5.5K Oct 17 01:51 readme.md
-rw-rw-rw-@ 1   19M Oct 17 15:09 v1.3.1-osx.zip

chmod +x GtR
./GtR
Loaded configuration text: 
{
  "CardShortSideInInches": 2.5,
  "CardLongSideInInches": 3.5,
  "BleedSizeInInches": 0.125, // This is (1.0f / 8.0f)
  "BorderPaddingInInches": 0.0625, // This is (1.0f / 16.0f)
  "SaveConfiguration": "Page" // Accepted values: Page or SingleImage
}

Cards will be 2.5 inches x 3.5 inches.
Cards will have a bleed size of 0.125 inches and margin of 0.0625 inches.
Saving images as pages of cards.
Creating images.
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: cHRM chunk does not match sRGB
libpng warning: iCCP: known incorrect sRGB profile
...
Saving images to /tmp/delete/images/20201017T151339.
Done!
Press any key to exit.
```
The warnings are printed for every card, I think.
The resulting images are corrupted as you can see
![MiscFront](https://user-images.githubusercontent.com/242252/96351777-acc4a880-108b-11eb-8abb-6e049b8f1300.png)

**Comment by ecm85** [comment]
Thanks for the report! We had actually found this already. 
We had found 3 issues on OSX. 
Version 3.1 fixed the first two, and so the SingleImage option should work now (I may have some tweaks to the font spacing still incoming).

The image corruption when doing Pages is still outstanding.

I don't actually have any idea what is causing that issue. 

It's worth noting that it does log those warnings even when doing SingleImage, which does work. 
So those warnings -might- be unrelated to the image corruption.

If you wanted to play around and see if you can figure out the issue, I'd love to hear any ideas. I don't have access to an OSX development environment and have just been using other users as testers. 

I've also got a couple reddit chats for this project, including one specifically for OSX. Ping me on reddit chat if you want to be included in them (username Storm1850).

**Comment by cgrinds** [comment]
I'll ping you there. I don't have a C# dev env setup to easily debug. Will check that. 

Agreed, the warnings don't matter - I used imagemagik to rewrite all the images - eliminating the warnings and reran - same issue.

The other thing I noticed with "Page" mode is the image corruption only happens with the pages that do a rotate or flip, (e.g. Rotate180FlipNone). MiscBack.png is good, MiscFront.png. SiteBack.png is good, SiteFront.png is bad. All the order cards are bad since rotates are used for the back and front.

**Comment by cgrinds** [comment]
Managed to get C# working and created a small test that rotates images.
This works correctly, the image is rotated and written without corruption
```
var bitmapL = new Bitmap("GtR/Images/Misc/JackQuill.png", true);
bitmapL.RotateFlip(RotateFlipType.Rotate90FlipXY);
bitmapL.Save($"/tmp/delete/images/foo/Pages/foo_L.png");
```

This does not work and creates the corrupted image.
```
var jackImageFront = imageCreator.CreateJackImageSword();
jackImageFront.Bitmap.RotateFlip(RotateFlipType.Rotate90FlipXY);
jackImageFront.Bitmap.Save("/tmp/delete/images/foo/Pages/fooJackImageFront.png");      
```
That means dot.net core is fine on the Mac and it's something in ImageCreator.

**Comment by cgrinds** [comment]
OK more progress. Have a reproducible fail and an idea on how to fix.
The crux is the order - if you draw and then rotate it fails.
If you rotate then draw it works.
Seems similar to [this](https://social.msdn.microsoft.com/Forums/vstudio/en-US/d1bf4005-d0bc-4e4d-9ea1-c307f1466ead/bug-in-bitmaprotateflip-and-graphicsdrawimage-in-visual-c), but still doesn't explain why it works on Windows but not Mac.

```
// Fails - most similar to what ImageCreator does
using(var img = Image.FromFile("/Users/cgrinds/Downloads/GtR/GtR/GtR/Images/Misc/JackSword.png")) {
    var bitmap = new Bitmap(821, 491);
    bitmap.SetResolution(300, 300);
    var graphics = Graphics.FromImage(bitmap);
    graphics.DrawImage(img, new Rectangle(0, 0, 821, 491)
    bitmap.RotateFlip(RotateFlipType.Rotate90FlipXY);
    bitmap.Save($"/tmp/delete/images/cbg/Pages/foo_TEST_1_1.png");
}
```

This works
```
// This works - Do the rotation then draw into a new bitmap
// rather than draw into a bitmap and then rotate
var bitmapSword = new Bitmap("/Users/cgrinds/Downloads/GtR/GtR/GtR/Images/Misc/JackSword.png", true);
bitmapSword.RotateFlip(RotateFlipType.Rotate90FlipXY);
Console.WriteLine($"w={bitmapSword.Width} h={bitmapSword.Height}");
var bitmap3 = new Bitmap(491, 821);
bitmap3.SetResolution(300, 300);

var graphics2 = Graphics.FromImage(bitmap3);
graphics2.DrawImage(bitmapSword, new Rectangle(0, 0, bitmap3.Width, bitmap3.Height));
bitmap3.Save($"/tmp/delete/images/cbg/Pages/foo_TEST_2_0.png");
```

**Comment by ecm85** [comment]
Wow great investigation so far! I'd guess a bug in the GDI+ mono port (much like the one I've already found that it doesn't handle unicode spaces correctly, like hair space, which is why I am now doing letter positioning by hand [ugh]).

**Comment by ecm85** [comment]
I'm going to investigate switching off of GDI+ entirely. It'll mean OSX won't need to install the port and should hopefully be smoother cross platform. Looking at https://github.com/dlemstra/Magick.NET or https://github.com/SixLabors/ImageSharp.

**Comment by ecm85** [comment]
Welp I am no longer doing letter positioning by hand. I figured out the unicode space issue, was not in fact a bug. Didn't really like either ImageSharp or Magick.NET. Going to look into the Pages issue next.

**Comment by ecm85** [comment]
Took a while, but I ported the tool to a website instead. It's now available at gtr.stormtide.net.

It's hosted in AWS on a Lambda, which means it's still using the mono port of gdi+. 

To fix the rotation issue, I found that if you add -extra- rotates, it works just fine. So now I have:
```
card.Bitmap.RotateFlip(RotateFlipType.Rotate180FlipNone);
card.Bitmap.RotateFlip(RotateFlipType.Rotate180FlipNone);
```

In there. It hurts but it works.

I fixed the other original issue by adding unicode points into the font i was using. I still had another layout bug, though - GDI+ on windows by default adds some padding that it appears not to in the mono port, so I had to manually account for that.

It's all working now tho.

Closing the issue. Let me know if you find anything else!
